Turn off telemetry in Docker images used in CI

amytai opened the pull request:
Using the new flag available in Docker images, turn off telemetry in Docker images used by Travis

amytai commented:
Related, what is `docker-run-tests.sh`? Can we remove it? Doesn't appear to be used anywhere.

amytai commented:
@cevian : https://github.com/timescale/timescaledb-docker/commit/9ba060636e2696d9dfc98929d78c588636a793d3

amytai commented:
@RobAtticus any word on `docker-run-tests.sh`? Can I delete?

erimatnor commented on scripts/docker-build.sh:
```diff
@@ -18,11 +18,11 @@ docker rm -f $(docker ps -a -q -f name=${BUILD_CONTAINER_NAME} 2>/dev/null) 2>/d
 if docker image ls -f "reference=${BUILD_IMAGE_NAME}:${PG_IMAGE_TAG}" | grep "$BUILD_IMAGE_NAME" 2>/dev/null; then
     echo "Using existing build image ${BUILD_IMAGE_NAME}:${PG_IMAGE_TAG}"
-    docker run -d --name ${BUILD_CONTAINER_NAME} -v ${BASE_DIR}:/src ${BUILD_IMAGE_NAME}:${PG_IMAGE_TAG}
+    docker run --env TIMESCALEDB_TELEMETRY=off -d --name ${BUILD_CONTAINER_NAME} -v ${BASE_DIR}:/src ${BUILD_IMAGE_NAME}:${PG_IMAGE_TAG}
```
Strictly, I don't think this needs the env variable since this is only a PostgreSQL build image. And, although TimescaleDB is later built in it, I don't think the extension is ever installed in the running container. I guess the env variable doesn't hurt though, although I am wondering if it is not better to add a line to the `postgresql.conf` file (like we do with `shared_preload_libraries`) while we are messing with the image anyway?

amytai replied:
Yes, that was how I turned off telemetry in `docker-build.sh` previously. But I removed it (and replaced with this env variable) for consistency, meaning, otherwise it wouldn't be clear to someone skimming the scripts why this particular `docker run` doesn't turn off telemetry; adding a line to `.conf` is buried deep in the `docker exec` line. I thought having both would be redundant, but I guess we could have both.

On a side note, I think I'll wait for your PR #778 to go in before rebase/merging this PR, since we are changing all the same files.

erimatnor replied:
I think the issue of having to pass this env variable to every place in every script that starts this image is very prone to errors. 

Setting this persistently in the image via the conf file only needs to be done at build time and we're done.

erimatnor commented on scripts/docker-build.sh:
```diff
@@ -40,4 +40,4 @@ docker rm -f ${BUILD_CONTAINER_NAME}
 echo
 echo "Built image ${IMAGE_NAME}:${TAG_NAME}"
-echo "Run 'docker run -d --name some-timescaledb -p 5432:5432 ${IMAGE_NAME}:${TAG_NAME}' to launch"
+echo "Run 'docker run --env TIMESCALEDB_TELEMETRY=off -d --name some-timescaledb -p 5432:5432 ${IMAGE_NAME}:${TAG_NAME}' to launch"
```
Again, I think it is much better and safer to set telemetry to off by default in the config file to not burden developers with this env variable. This also avoids having this env variable trickle down into other scripts that depend on the built image.

svenklemm commented:
> @cevian : [timescale/timescaledb-docker@9ba0606](https://github.com/timescale/timescaledb-docker/commit/9ba060636e2696d9dfc98929d78c588636a793d3)

This will only disable telemetry when the environment variable is set while initdb is run, which is ok to disable it for tests but for normal operations it might be misleading because the environment variable does nothing once the cluster is initialized, we should at least document that behaviour if its intended.

amytai commented:
@svenklemm : not sure what you're referencing. The env variable is written to `postgresql.conf`. I've connected to the docker image after initdb + all the entrypoint scripts run, and `timescaledb.telemetry_level` is set correctly. I have also stopped/started the docker image, and `telemetry_level` is still set to off. What's the concern?

svenklemm commented:
@amytai My comment was not about the PR but the environment variable in general. If the variable is not set when initdb is run it will have no effect. If you have your cluster initialized and start the container with the bind/volume mount of the initialized postgres data directory it will have no effect.

amytai commented:
@svenklemm : Ah yes, that is correct, since the env variable is only set inside the init scripts. That's all I intended anyway, since if your cluster is up and running, you should just be able to connect via `psql` and set the GUC yourself (or otherwise follow our instructions in the docs for disabling telemetry). I added this env var precisely for the init case, when you want users to have the option to disable telemetry before it pings for the first time. 
Will patch appropriate README.

erimatnor commented on scripts/docker-build.sh:
```diff
@@ -31,7 +31,7 @@ else
 
 # Build and install the extension with debug symbols and assertions
-docker exec -u root -it ${BUILD_CONTAINER_NAME} /bin/bash -c "cp -a /src/{src,sql,test,scripts,version.config,CMakeLists.txt,timescaledb.control.in} /build/ && cd build/debug && cmake -DCMAKE_BUILD_TYPE=Debug .. && make && make install && echo \"shared_preload_libraries = 'timescaledb'\" >> /usr/local/share/postgresql/postgresql.conf.sample && echo \"timescaledb.telemetry_level=off\" >> /usr/loca ...
+docker exec -u root -it ${BUILD_CONTAINER_NAME} /bin/bash -c "cp -a /src/{src,sql,test,scripts,version.config,CMakeLists.txt,timescaledb.control.in} /build/ && cd build/debug && cmake -DCMAKE_BUILD_TYPE=Debug .. && make && make install && echo \"shared_preload_libraries = 'timescaledb'\" >> /usr/local/share/postgresql/postgresql.conf.sample && cd / && rm -rf /build"
```
This was a much better approach.

amytai commented:
@erimatnor : I just updated to only pass the flag to Timescale-built docker images. Also reverted the logic in `docker-build.sh` that sets `telemetry_level` in the conf file. Can you take a look again?
